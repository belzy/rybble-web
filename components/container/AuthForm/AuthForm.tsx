import React from 'react';
import {
  VStack,
} from '@chakra-ui/react';
import { TAuthFormProps } from './AuthForm.types';

/**
 * Used to wrap auth form inputs.
 * @param { React.ReactNode } children 
 * @returns { React.ReactElement }
 * @example <AuthForm> //...form elements </AuthForm>
 */
const AuthForm = ({ children }
  : TAuthFormProps)
  : React.ReactElement<TAuthFormProps> => {

  return (
    <form
      data-testid='auth-form'
      style={{ marginTop: '64px', width: '100%' }}
    >
      <VStack spacing={ 8 }>

        { children }

      </VStack>
    </form>
  );
};

export default AuthForm;