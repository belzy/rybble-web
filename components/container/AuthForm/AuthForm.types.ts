import React from 'react';

export type TAuthFormProps = {
  children: React.ReactNode
};