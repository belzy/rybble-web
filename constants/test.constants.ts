
type TTest = {
  SNAPSHOT_STRINGS: Array<string>
};

/**
 * Message strings to be used with testing functions like 'describe' and 'it'.
 * 
 * @example 
 * it(SNAPSHOT_STRINGS[0], () => {
 * 
 * // Your Test...
 * 
 * })
 */
export const TEST_STRINGS: TTest = {
  SNAPSHOT_STRINGS: [
    'should match snapshot'
  ]

};