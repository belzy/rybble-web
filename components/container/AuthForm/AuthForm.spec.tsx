import AuthForm from './AuthForm';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { TEST_STRINGS } from '../../../constants/test.constants';

const { SNAPSHOT_STRINGS } = TEST_STRINGS;

describe('components/AuthForm', () => {

  it(SNAPSHOT_STRINGS[0], () => {

    const { container } = render(<AuthForm><p>Test</p></AuthForm>);

    expect(container).toMatchSnapshot();

  });

  it('should render a child', () => {

    render(
      <AuthForm>
        <p>Should Render</p>
      </AuthForm>
    );

    const formEl = screen.getByTestId('auth-form');

    expect(formEl.children.length).toEqual(1);

  });

});

