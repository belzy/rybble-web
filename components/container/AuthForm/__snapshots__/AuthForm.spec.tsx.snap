// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AuthForm should match snapshot 1`] = `
<div>
  <form
    data-testid="auth-form"
    style="margin-top: 64px; width: 100%;"
  >
    <div
      class="chakra-stack css-14witgw"
    >
      <p>
        Test
      </p>
    </div>
  </form>
</div>
`;
